girder-dandi-archive/girder_dandi_archive/__init__.py

```diff
@@ -19,6 +20,7 @@ def load(self, info):
             {"$setOnInsert": {"value": 1}},
             upsert=True,
         )
+        Folder().ensureIndex(([("meta.dandiset.identifier", 1), ("parentId", 1)], {}))
```

Comment:
```suggestion
        Folder().ensureIndex([("meta.dandiset.identifier", 1), ("parentId", 1)])
```
Since there are no options, there's no need to pass a tuple: https://github.com/girder/girder/blob/355e85b40bc86e37147a702aabf8338e2a13f9bb/girder/models/model_base.py#L196

Reply:
That code seems to make the assumption that all lists/tuples are of the form `(index, options)`, and all strings are simply the name of the index. I need to create a compound index, which has to be encoded as a list/tuple. That code will misinterpret my compound index and complain that the options argument is malformed. The only way I found to avoid the misunderstanding is by specifying the empty options.

If I were doing `Folder().ensureIndex(("meta.dandiset.identifier", {}))`, you would be absolutely right.

Reply:
Ah, I see. You're right, thanks.